pandas/core/frame.py

```diff
@@ -6243,13 +6243,24 @@ def explode(self, column: Union[str, Tuple]) -> "DataFrame":
         if not self.columns.is_unique:
             raise ValueError("columns must be unique")
 
-        result = self[column].explode()
-        return (
+        column_with_index = self[column].reset_index()
+        result = (
             self.drop([column], axis=1)
-            .join(result)
-            .reindex(columns=self.columns, copy=False)
+            .reset_index()
+            .join(column_with_index[column].explode())
```

Comment:
I'm thinking that all this is the same as:
```python
exploded_col = self.pop(column).reset_index(drop=True).explode()
self.reset_index().join(exploded_col)
```

We probably don't want to modify self but a copy, because we need `self.columns` later, but I think that is a bit clearer.